### PR TITLE
[POT] Fix get layout from model

### PR DESCRIPTION
**Pull request by IndiraSalyahova**
### Tickets:
 - 71869

### CODE REVIEW

File: tools/pot/openvino/tools/pot/data_loaders/image_loader.py [hunk 1]
```diff
@@ -54,9 +58,11 @@ def get_layout(self, input_node=None):
             self._layout = Layout(self._layout)
             return
 
-        if input_node:
-            layout_from_ir = input_node.graph.graph.get('layout', None)
+        if input_node and input_node.graph.meta_data.get('layout', None) not in [None, '()']:
+            layout_from_ir = get_layout_values(input_node.graph.meta_data.get('layout', None))
```
Comment (KodiaqQ) [inline tools/pot/openvino/tools/pot/data_loaders/image_loader.py:62]: Not sure, but a number of incoming models can be without the `meta_data` section. I'm talking about models that were serialized using runtime methods (GNA, for example). Maybe we just need to get some kind of confirmation that `meta_data` will exists always. cc @AlexKoff88

File: tools/pot/openvino/tools/pot/engines/ie_engine.py [hunk 2]
```diff
@@ -233,7 +233,12 @@ def _fill_input(self, model, image_batch):
...
             input_blob_name = self._get_input_any_name(input_blob)
-            return {input_blob_name: np.stack(image_batch, axis=0)}
+            image_batch = {input_blob_name: np.stack(image_batch, axis=0)}
+            if Shape(image_batch[input_blob_name].shape) != input_info[0].shape:
+                raise ValueError(f"Incompatible input shapes. "
+                                 f"Cannot infer {Shape(image_batch[input_blob_name].shape)} into {input_info[0].shape}."
+                                 "Try to specify the layout of the model.")
+            return image_batch
 
         if len(input_info) == 2:
```
Comment (KodiaqQ) [inline tools/pot/openvino/tools/pot/engines/ie_engine.py:243]: I think that we should check Shape's here too. This condition is actual for faster-/mask-rcnn models.

File: tools/pot/openvino/tools/pot/data_loaders/image_loader.py (hunk 1 above)
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/data_loaders/image_loader.py:62]: @KodiaqQ, does your commnet relevant to the image loader? As far as I know, meta-information is saved during the serialization. But probably someone from the ngraph team should comment here: @ilyachur, @GlebKazantaev?

Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/data_loaders/image_loader.py:64]: I suggest accepting this a wait for the validation of Simplified mode on a wide range of models

**Review by KodiaqQ — APPROVED** [review]

File: tools/pot/openvino/tools/pot/data_loaders/image_loader.py (hunk 1 above)
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/data_loaders/image_loader.py:64]: Per discussion with @achetver, it seems that this is a valid case and we should check the availability of meta info in the model before trying to access it. @IndiraSalyahova, can you please change the code accordingly?

Comment (IndiraSalyahova) [reply tools/pot/openvino/tools/pot/data_loaders/image_loader.py:64]: @AlexKoff88 this condition was added. `and hasattr(input_node.graph, 'meta_data') `

**Review by AlexKoff88 — APPROVED** [review]

**Pull request merged**